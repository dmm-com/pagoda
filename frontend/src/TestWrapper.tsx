import { createTheme, ThemeProvider } from "@mui/material/styles";
import { SnackbarProvider } from "notistack";
import React, { FC } from "react";
import { MemoryRouter } from "react-router";

export const TestWrapper: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        <MemoryRouter>{children}</MemoryRouter>
      </SnackbarProvider>
    </ThemeProvider>
  );
};

export const TestWrapperWithoutRoutes: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        {children}
      </SnackbarProvider>
    </ThemeProvider>
  );
};
