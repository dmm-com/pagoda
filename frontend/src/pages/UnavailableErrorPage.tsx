import { Box, Button, Typography } from "@mui/material";
import { FC, useCallback } from "react";

import { topPath } from "../routes/Routes";

export const UnavailableErrorPage: FC = () => {
  const handleClickGoToTop = useCallback(() => {
    location.href = topPath();
  }, []);

  return (
    <Box
      display="flex"
      flexDirection="column"
      flexGrow={1}
      alignItems="center"
      justifyContent="center"
    >
      <Box display="flex" my="52px">
        <Typography
          id="sorry_forbidden"
          variant="h1"
          color="#B0BEC5"
          fontWeight="bold"
        >
          利用できません:;(∩´﹏`∩);:
        </Typography>
      </Box>
      <Box display="flex" flexDirection="column" alignItems="center">
        <Typography color="#455A64">
          このページは現在、利用ができません。
        </Typography>
        <Typography color="#455A64">
          管理者からのお知らせをご覧いただくか、お問合せください。
        </Typography>
      </Box>
      <Box>
        <Button
          variant="contained"
          color="secondary"
          sx={{ borderRadius: "16px", my: "40px" }}
          onClick={handleClickGoToTop}
        >
          トップページへ
        </Button>
      </Box>
    </Box>
  );
};
