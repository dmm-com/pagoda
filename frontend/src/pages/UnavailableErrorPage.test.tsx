/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";

import { UnavailableErrorPage } from "./UnavailableErrorPage";

import { TestWrapper } from "TestWrapper";

afterEach(() => {
  jest.clearAllMocks();
});

test("should match snapshot", async () => {
  // wait async calls and get rendered fragment
  const result = render(<UnavailableErrorPage />, {
    wrapper: TestWrapper,
  });

  expect(result).toMatchSnapshot();
});
