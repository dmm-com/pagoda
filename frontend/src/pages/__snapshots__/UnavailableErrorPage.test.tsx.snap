// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-i80i99"
      >
        <div
          class="MuiBox-root css-185kxi8"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-qgkryb-MuiTypography-root"
            id="sorry_forbidden"
          >
            利用できません:;(∩´﹏\`∩);:
          </h1>
        </div>
        <div
          class="MuiBox-root css-dvxtzn"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-e679fg-MuiTypography-root"
          >
            このページは現在、利用ができません。
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-e679fg-MuiTypography-root"
          >
            管理者からのお知らせをご覧いただくか、お問合せください。
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mkkyro-MuiButtonBase-root-MuiButton-root"
            id=":r0:"
            tabindex="0"
            type="button"
          >
            トップページへ
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-i80i99"
    >
      <div
        class="MuiBox-root css-185kxi8"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-qgkryb-MuiTypography-root"
          id="sorry_forbidden"
        >
          利用できません:;(∩´﹏\`∩);:
        </h1>
      </div>
      <div
        class="MuiBox-root css-dvxtzn"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-e679fg-MuiTypography-root"
        >
          このページは現在、利用ができません。
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-e679fg-MuiTypography-root"
        >
          管理者からのお知らせをご覧いただくか、お問合せください。
        </p>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1mkkyro-MuiButtonBase-root-MuiButton-root"
          id=":r0:"
          tabindex="0"
          type="button"
        >
          トップページへ
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
